### ISSUE TITLE
URI deeplink not working on Cold Start

### ISSUE THREAD

**Issue by NassB** [RodrigoSMarques/flutter_branch_sdk#62]
I am trying to open an iOS application with a URI deeplink (scheme://..........) to redirect users to a specific state at the launch. It is not working properly for one case : when the app has been freshly installed on the device. App is launched through the URI deeplink but deeplink data is not detected. On next app launches, everything works fine.
The issue occurs with the plugin example too.
I can't use Universal link for this specific use case.
Note that there is no issue with Android devices.

Flutter version 1.22.3
flutter_branch_sdk version 1.2.0

Thanks

**Comment by RodrigoSMarques** [comment]
Hi @NassB 
Testing on iOS is always more complex.
You need to use the Test branch_key in INFO.PLIST and register the device as tests.
You can get more information here:
https://help.branch.io/developers-hub/docs/ios-testing
https://help.branch.io/developers-hub/docs/ios-testing
https://help.branch.io/developers-hub/docs/ios-troubleshooting#app-not-passing-data

The sample application also has this problem because it has the production key.

I have published applications and I know of other people who have them and despite this problem, it works normally after published.

**Comment by RodrigoSMarques** [comment]
Closed. No activity in the last 15 days.
If necessary open again.

**Comment by nizo-m** [comment]
Hi @NassB 
I'm facing the same issue, were you able to fix it ?

**Comment by NassB** [comment]
> Hi @NassB
> I'm facing the same issue, were you able to fix it ?

Hi @nizo-m, I haven't returned to the problem since December, so I'm going to recheck my application (hopefully soon) and I'll keep you informed of any improvements or if I manage to fix the problem.

**Comment by nizo-m** [comment]
> > Hi @NassB
> > I'm facing the same issue, were you able to fix it ?
> 
> Hi @nizo-m, I haven't returned to the problem since December, so I'm going to recheck my application (hopefully soon) and I'll keep you informed of any improvements or if I manage to fix the problem.

Thanks @NassB

**Comment by RodrigoSMarques** [comment]
Hello guys.
In ios to work properly you must use the test key and Add a Test Device to Your Branch Account

https://help.branch.io/developers-hub/docs/ios-testing#use-test-key
https://help.branch.io/developers-hub/docs/ios-testing#simulate-an-install

**Comment by nizo-m** [comment]
> Hello guys.
> In ios to work properly you must use the test key and Add a Test Device to Your Branch Account
> 
> https://help.branch.io/developers-hub/docs/ios-testing#use-test-key
> https://help.branch.io/developers-hub/docs/ios-testing#simulate-an-install

Thank you @RodrigoSMarques 
I will try what you suggested.

**Comment by nizo-m** [comment]
@RodrigoSMarques 
Should I simulate an install if the app is already live ?

**Comment by AristideVB** [comment]
@RodrigoSMarques We're also encountering this issue with URI deeplinks not working on cold starts when the app has been freshly installed. Based on our understanding, this seems to be related to the initialization sequence of the Branch SDK in conjunction with the iOS lifecycle. During a cold start, the Branch SDK is initialized in application(_:didFinishLaunchingWithOptions:) and might handle the deeplink before the Flutter routing logic is ready. This can result in the deeplink data not being detected on the first launch post-installation.

This appears to be a synchronization issue between the Branch SDK's deeplink processing and the Flutter app's state setup. Ensuring that the deeplink data is handled and available to Flutter immediately after initialization might help address this issue.

**Comment by RodrigoSMarques** [comment]
Take a test with version 8.x that was released last week.

**Comment by AristideVB** [comment]
@RodrigoSMarques I confirm upgrading to `flutter_branch_sdk: ^8.0.2` resolved the issue. 

Thanks for your hard work ⚡️⚡️

Can you provide some details on what the issue was & how you solved it ? Thank you very much

**Comment by RodrigoSMarques** [comment]
From version 6.x to version 7.x I included a delay in the SDK initialization to allow switching via the production and live key code. 

The issues identified that this delay was causing data loss (although in my tests I was never able to reproduce this problem). 

In version 8.x I returned the code to the same way it worked in 6.x, that is, the Branch SDK initializing in the native layer, even before the dart/flutter code is executed.
